octopusdeploy_framework/resource_lifecycle.go

```diff
@@ -0,0 +1,362 @@
+package octopusdeploy_framework
+
+import (
+	"context"
+	"fmt"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/lifecycles"
+	"github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy_framework/schemas"
+	"github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy_framework/util"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"strings"
+)
+
+type lifecycleTypeResource struct {
+	*Config
+}
+
+type lifecycleTypeResourceModel struct {
+	ID                      types.String `tfsdk:"id"`
+	SpaceID                 types.String `tfsdk:"space_id"`
+	Name                    types.String `tfsdk:"name"`
+	Description             types.String `tfsdk:"description"`
+	Phase                   types.List   `tfsdk:"phase"`
+	ReleaseRetentionPolicy  types.List   `tfsdk:"release_retention_policy"`
+	TentacleRetentionPolicy types.List   `tfsdk:"tentacle_retention_policy"`
+}
+
+func NewLifecycleResource() resource.ResourceWithImportState {
```

Comment:
The `ResourceWithImportState` is not required here
```suggestion
func NewLifecycleResource() resource.Resource {
```